pkg/gatherers/conditional/common_test.go

```diff
@@ -71,3 +73,62 @@ func Test_getAlertPodNamespace(t *testing.T) {
 		})
 	}
 }
+
+// nolint:dupl
+func TestGetAlertPodContainer(t *testing.T) {
+	testCases := []struct {
+		name           string
+		labels         AlertLabels
+		expectedResult string
+		expectedErr    error
+	}{
+		{
+			name: "valid container label",
+			labels: AlertLabels{
+				"container": "my-container",
+			},
+			expectedResult: "my-container",
+			expectedErr:    nil,
+		},
+		{
+			name:           "missing container label",
+			labels:         AlertLabels{},
+			expectedResult: "",
+			expectedErr:    ErrAlertPodContainerMissing,
+		},
+		{
+			name: "empty container label",
+			labels: AlertLabels{
+				"container": "",
+			},
+			expectedResult: "",
+			expectedErr:    ErrAlertPodContainerMissing,
+		},
+		{
+			name:           "nil labels",
+			labels:         nil,
+			expectedResult: "",
+			expectedErr:    ErrAlertPodContainerMissing,
+		},
+		{
+			name: "valid labels but missing container key",
+			labels: AlertLabels{
+				"namespace": "my-namespace",
+			},
+			expectedResult: "",
+			expectedErr:    ErrAlertPodContainerMissing,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			container, err := getAlertPodContainer(tc.labels)
+			if container != tc.expectedResult {
+				t.Errorf("Expected result to be %s but got %s", tc.expectedResult, container)
+			}
+			if err != tc.expectedErr {
+				t.Errorf("Expected error to be %v but got %v", tc.expectedErr, err)
+			}
```

Comment:
nit: we can use the `assert` library when it's already in our deps.

Reply:
I guess that I tried to use it, but CI was complaining about the dependency... But, I guess that I can check the dependency and try again.